### DISCUSSION TITLE
How to handle gene duplication and loss properly

### DISCUSSION THREAD

**Discussion by iacopogentile** [hahnlab/CAGEE#36] [Q&A] [answered]
Hi,

CAGEE seems an incredibly interesting tool and the related published work is very interesting, congratulations to all the authors.
I was just wondering what is the best way to handle gene duplication and loss in the input data. is it suggested to identify the 'ancestral' copy that will have data for all the species, while the duplicated one will have lineage-specific data, and empty values for those species where the speciation event happened before the duplication event?
Best,
IG

**Comment by benfulton** [comment, marked as answer]
Good question. There isn't anything in CAGEE that addresses duplicates directly. Dr. Hahn thinks it would be best to simply not include them.

**Reply by iacopogentile** [reply]
Thank you! what about gene loss (even in singleton genes)? can CAGEE accept empty values in the species columns, or should you prune the species where the gene is missing out of the tree ?

**Reply by benfulton** [reply]
Missing values are allowed in the input file. Represent them with the single characters "?", "-", or "N".

**Reply by iacopogentile** [reply]
Thank you!
